sympy/functions/combinatorial/factorials.py

```diff
@@ -848,15 +854,22 @@ def eval(cls, n, k):
         d = n - k
         if d.is_zero or k.is_zero:
             return S.One
-        elif d.is_zero is False:
-            if (k - 1).is_zero:
-                return n
-            elif k.is_negative:
+        if (k - 1).is_zero:
+            return n
+        if k.is_integer:
+            if k.is_negative:
                 return S.Zero
-            elif n.is_integer and n.is_nonnegative and d.is_negative:
+            if n.is_integer and n.is_nonnegative and d.is_negative:
                 return S.Zero
-        if k.is_Integer and k > 0 and n.is_Number:
-            return cls._eval(n, k)
+            if n.is_Number:
```

Comment:
I could have used here too?

Reply:
`_eval` seems to expect that `n.is_Number` is true.

Reply:
But if `n` is a complex, it seems `_eval` work fine.